.github/workflows/my-ci-pipeline.yml

```diff
@@ -27,4 +27,21 @@ jobs:
           channel-id: 'U01U1N568MT'
           slack-message: "GitHub build result: ${{ job.status }}\n${{ github.event.pull_request.html_url || github.event.head_commit.url }}"
         env:
-          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
\ No newline at end of file
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+  push-prod:
+    name: Push production image to Docker Hub
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Build and push
+      env:
+          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
+      run: |
+        echo ${{ secrets.DOCKER_HUB_PASSWORD }} | docker login --username $DOCKER_HUB_USERNAME --password-stdin     
+        docker build --target production --tag $DOCKER_HUB_USERNAME/todo-app:$GITHUB_SHA --tag $DOCKER_HUB_USERNAME/todo-app:main .
+        docker push $DOCKER_HUB_USERNAME/todo-app --all-tags

```

Comment:
Personal preference, but I'm a fan of keeping commands separate by default, to make it easier to debug issues & break the logs up, so I'd lean towards having three steps here rather than one

Reply:
Entirely separately, nice approach handling both the tags, that's a nice feature here 👍